clients/src/main/java/org/apache/kafka/common/metrics/QuotaViolationException.java

```diff
@@ -17,15 +17,38 @@
 package org.apache.kafka.common.metrics;
 
 import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.MetricName;
 
 /**
  * Thrown when a sensor records a value that causes a metric to go outside the bounds configured as its quota
  */
 public class QuotaViolationException extends KafkaException {
 
     private static final long serialVersionUID = 1L;
+    private final MetricName metricName;
+    private final double value;
+    private final double bound;
 
-    public QuotaViolationException(String m) {
-        super(m);
+    public QuotaViolationException(MetricName metricName, Double value, Double bound) {
```

Comment:
Nitpick: `value` and `bound` should be lowercase `double` since they can't be null (annoying that Scala and Java are different in this respect so it's a bit confusing when writing code in both languages).